File: data-access-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__CredentialStoringDataAccessProvider.java
package ${package}.${pluginName};

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import se.curity.identityserver.sdk.Nullable;
import se.curity.identityserver.sdk.attribute.SubjectAttributes;
import se.curity.identityserver.sdk.datasource.CredentialDataAccessProviderFactory;
import se.curity.identityserver.sdk.datasource.CredentialStoringDataAccessProvider;

public final class ${pluginName}CredentialStoringDataAccessProvider implements CredentialStoringDataAccessProvider, CredentialDataAccessProviderFactory
{
    private static final Logger _logger = LoggerFactory.getLogger(${pluginName}CredentialDataAccessProvider.class);

    private final ${pluginName}DataAccessProviderConfig _configuration;

    @SuppressWarnings("unused") // used through DI
    public ${pluginName}CredentialStoringDataAccessProvider(${pluginName}DataAccessProviderConfig configuration)
    {
        _configuration = configuration;
    }

    @Override
    @Nullable
    public GetResult get(SubjectAttributes subjectAttributes)
    {
        _logger.debug("Received request to retrieve a credential for a given subject: {}", subjectAttributes.getSubject());

        throw new UnsupportedOperationException();
    }

    @Override
    public void store(SubjectAttributes subjectAttributes, String password)
    {
        _logger.debug("Received request to store password for user: {}", subjectAttributes.getSubject());
        throw new UnsupportedOperationException();
    }

    @Override
    public boolean delete(SubjectAttributes subjectAttributes) {
        _logger.debug("Received request to delete credentials for user: {}", subjectAttributes.getSubject());

        throw new UnsupportedOperationException();
    }
}

File: kotlin-data-access-provider-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__CredentialStoringDataAccessProvider.kt
package ${package}.${pluginName}

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import se.curity.identityserver.sdk.Nullable
import se.curity.identityserver.sdk.attribute.SubjectAttributes
import se.curity.identityserver.sdk.datasource.CredentialDataAccessProviderFactory
import se.curity.identityserver.sdk.datasource.CredentialStoringDataAccessProvider
import se.curity.identityserver.sdk.datasource.CredentialStoringDataAccessProvider.GetResult

class ${pluginName}CredentialStoringDataAccessProvider(private val _configuration: ${pluginName}DataAccessProviderConfig): CredentialStoringDataAccessProvider, CredentialDataAccessProviderFactory
{

    @Nullable
    override fun get(subjectAttributes: SubjectAttributes): GetResult? {
        _logger.debug("Received request to update password for user : {}", subjectAttributes.subject)

        TODO()
    }

    override fun delete(subjectAttributes: SubjectAttributes): Boolean {
        _logger.debug("Received request to delete credentials for user: {}", subjectAttributes.subject)

        TODO()
    }

    override fun store(subjectAttributes: SubjectAttributes, password: String?) {
        _logger.debug("Received request to store password for user : {}", subjectAttributes.subject)

        TODO()
    }

    companion object
    {
        private val _logger: Logger = LoggerFactory.getLogger(${pluginName}CredentialStoringDataAccessProvider::class.java)
    }
}
